versioned/persist/mongodb/src/test/java/org/projectnessie/versioned/persist/mongodb/TestMongoDatabaseAdapter.java

```diff
@@ -0,0 +1,114 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import de.flapdoodle.embed.mongo.Command;
+import de.flapdoodle.embed.mongo.MongodExecutable;
+import de.flapdoodle.embed.mongo.MongodStarter;
+import de.flapdoodle.embed.mongo.config.Defaults;
+import de.flapdoodle.embed.mongo.config.MongodConfig;
+import de.flapdoodle.embed.mongo.config.MongodProcessOutputConfig;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import de.flapdoodle.embed.process.config.io.ProcessOutput;
+import de.flapdoodle.embed.process.io.StreamProcessor;
+import de.flapdoodle.embed.process.runtime.Network;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.projectnessie.versioned.persist.tests.AbstractTieredCommitsTest;
+
+public class TestMongoDatabaseAdapter extends AbstractTieredCommitsTest {
+  private static final Pattern LISTEN_ON_PORT_PATTERN =
+      Pattern.compile(
+          ".*NETWORK ([^\\n]*) waiting for connections on port ([0-9]+)\\n.*",
+          Pattern.MULTILINE | Pattern.DOTALL);
+
+  private static MongodExecutable mongo;
+
+  @BeforeAll
+  static void startMongo() throws IOException {
+    AtomicInteger port = new AtomicInteger();
+
+    ProcessOutput defaultOutput = MongodProcessOutputConfig.getDefaultInstance(Command.MongoD);
+    StreamProcessor capturedStdout =
+        new StreamProcessor() {
+          private final StringBuilder buffer = new StringBuilder();
+
+          @Override
+          public void process(String block) {
+            if (port.get() == 0) {
+              buffer.append(block);
+              Matcher matcher = LISTEN_ON_PORT_PATTERN.matcher(buffer);
+              if (matcher.matches()) {
+                String portString = matcher.group(2);
+                port.set(Integer.parseInt(portString));
+              }
+            }
+            defaultOutput.getOutput().process(block);
+          }
+
+          @Override
+          public void onProcessed() {
+            defaultOutput.getOutput().onProcessed();
+          }
+        };
+
+    MongodStarter starter =
+        MongodStarter.getInstance(
+            Defaults.runtimeConfigFor(Command.MongoD)
+                .processOutput(
+                    new ProcessOutput(
+                        capturedStdout, defaultOutput.getError(), defaultOutput.getCommands()))
+                .build());
+
+    MongodConfig mongodConfig =
+        MongodConfig.builder()
+            .version(Version.Main.PRODUCTION)
+            .net(new Net(0, Network.localhostIsIPv6()))
+            .build();
+
+    mongo = starter.prepare(mongodConfig);
+    mongo.start();
+
+    assertThat(port).hasPositiveValue();
+
+    createAdapter(
+        "MongoDB",
+        config -> {
+          MongoDatabaseAdapterConfig testConfig =
```

Comment:
Better move this into `MongoDatabaseAdapter.<init>()` if `config.getClient()` returns `null`.
It's not the greatest approach, but let's keep the functionality the same across all adapter implementations.

Reply:
That way it's also guaranteed that the `MongoDatabaseClient` is closed when the `MongoDatabaseAdpater` is closed and not left behind when a test finishes.

Reply:
The closing of `MongoDatabaseClient` is handled by reference counting already. Will move code for consistency anyway.

Reply:
Added default client creation logic to the adapter in a1c226a4c4965e21b3c581d638a8fc302d8a4061